client/src/common/components/Filter.jsx

```diff
@@ -22,14 +25,29 @@ class Filter extends PureComponent {
     this.debouncedFilter.cancel();
   }
 
-  filter = query => {
-    const { options, onFilter } = this.props;
+  filterByFields = query => {
+    const { filterBy, options } = this.props;
+    const match = new RegExp(query, 'i');
```

Comment:
Please use case-invariant (toLowerCase()/toLocaleLowerCase) `String.includes` search as regexp constructor requires proper escaping (try "*" as a search query) and can lead to REDOS.